### ISSUE TITLE
Error when using `out` parameter

### ISSUE THREAD

**Issue by keesey** [ivogabe/gulp-typescript#86]
I'm trying to switch to using the new `out` parameter, but I get this error when I try to generate JavaScript files:

```
TypeError: Cannot read property 'referencedFiles' of undefined
    at sortedEmit (E:\Git\JS.Projects\wfcf-utils\node_modules\gulp-typescript\release\project.js:414:40)
    at Project.compile (E:\Git\JS.Projects\wfcf-utils\node_modules\gulp-typescript\release\project.js:427:17)
    at Object.callback (E:\Git\JS.Projects\wfcf-utils\node_modules\gulp-typescript\release\main.js:63:32)
    at E:\Git\JS.Projects\wfcf-utils\node_modules\gulp-typescript\release\project.js:250:37
    at fs.js:266:14
    at E:\Git\JS.Projects\wfcf-utils\node_modules\karma\node_modules\graceful-fs\graceful-fs.js:104:5
    at Object.oncomplete (fs.js:107:15)
```

If I change the line (`project.js`, line 414) that the error occurs in to this, it works:

```
               var references = tsFile ? tsFile.referencedFiles.map(function (file) { return Project.getFileName(file); }) : [];
```

(Not sure if that's the best fix, though.)

**Comment by keesey** [comment]
(While I'm at it, some more documentation on the `out` parameter would be nice. I could only tell what the string was supposed to represent by looking at the test code.)

**Comment by ivogabe** [comment]
Can you post your gulpfile?

**Comment by ivogabe** [comment]
Can you post an example to reproduce this? `out` works for me, but there may be something special with your setup.

**Comment by keesey** [comment]
I've created a small sample project that replicates the issue.

Files:
```
bower.json
gulpfile.js
package,json
scripts/
  app.ts
  functions/
    jquery.ts
```

`bower.json`:
```
{
  "name": "test-gulp-typescript",
  "version": "0.0.0",
  "main": [
    "./dist/main.js"
  ],
  "dependencies": {
    "jquery": "~2.1.3"
  },
  "devDependencies": {
    "dt-jquery": "~2.0.0"
  },
  "private": true
}
```

`gulpfile.js`:
```
'use strict';

var gulp = require('gulp');

var merge = require('merge2');
var typescript = require('gulp-typescript');

var TYPESCRIPT_PROJECT = typescript.createProject({
  declarationFiles: true,
  noImplicitAny: true,
  out: 'main.js',
  sortOutput: true,
  sourceRoot: '',
  target: 'ES3'
});

gulp.task('scripts', function() {
  var tsResult = gulp
    .src("scripts/**/*.ts")
    .pipe(typescript(TYPESCRIPT_PROJECT));
  return merge(
    tsResult.dts
      .pipe(gulp.dest('dist')),
    tsResult.js
  	  .pipe(gulp.dest('dist'))
  );
});

gulp.task('default', ['scripts']);
```

`package.json`:
```
{
  "name": "test-gulp-typescript",
  "version": "0.0.0",
  "dependencies": {},
  "devDependencies": {
    "gulp": "^3.8.11",
    "typescript": "^1.5.0-alpha",
    "gulp-typescript": "^2.5.0",
    "merge2": "^0.3.2"
  },
  "engines": {
    "node": ">=0.8.0"
  }
}
```

`scripts/app.ts`:
```
/// <reference path="../bower_components/dt-jquery/jquery.d.ts" />

'use strict';
```

`scripts/functions/jquery.ts`:
```
/// <reference path="../app.ts" />

module test.gulp_typescript {
  export function jquery() {
    return $.trim('  use a JQuery function  ');
  }
}
```

**Comment by englercj** [comment]
+1 Same issue here with v2.6.0.

**Comment by ivogabe** [comment]
This should be fixed in #95
